### PR TITLE
DPL2Decoder cleanup

### PR DESCRIPTION
**Pull request by MerryMage**
Just cleaning up, no behaviour change.

### CODE REVIEW

File: Source/Core/AudioCommon/DPL2Decoder.cpp [hunk 1]
```diff
@@ -35,33 +35,24 @@ static float adapt_l_gain, adapt_r_gain, adapt_lpr_gain, adapt_lmr_gain;
...
 static float LFE_buf[256];
 static unsigned int lfe_pos;
-static float* filter_coefs_lfe;
+static std::vector<float> filter_coefs_lfe;
 static unsigned int len125;
 
-template <class T, class _ftype_t>
-static _ftype_t DotProduct(int count, const T* buf, const _ftype_t* coefficients)
+template <class T>
+static float DotProduct(int count, const T* buf, const std::vector<float>& coeffs, int offset)
```
Comment (lioncash) [inline Source/Core/AudioCommon/DPL2Decoder.cpp]: Wouldn't this function be able to be replaced/simplified with `std::inner_product`?

File: Source/Core/AudioCommon/DPL2Decoder.cpp [hunk 2]
```diff
@@ -96,18 +86,22 @@ static T FIRFilter(const T* buf, int pos, int len, int count, const float* coeff
...
-    *w++ = float(0.54 - 0.46 * cos(k * (float)i));
+    w[i] = static_cast<float>(0.54 - 0.46 * cos(k * i));
+
+  return w;
 }
 
 /******************************************************************************
-*  FIR filter design
-******************************************************************************/
+ *  FIR filter design
```
Comment (lioncash) [inline Source/Core/AudioCommon/DPL2Decoder.cpp]: The excessive comment header and footer can just be replaced with `// FIR filter design`

File: Source/Core/AudioCommon/DPL2Decoder.cpp [hunk 3]
```diff
@@ -96,18 +86,22 @@ static T FIRFilter(const T* buf, int pos, int len, int count, const float* coeff
 // n window length
 // w buffer for the window parameters
```
Comment (lioncash) [inline Source/Core/AudioCommon/DPL2Decoder.cpp]: This comment line should be removed now.

File: Source/Core/AudioCommon/DPL2Decoder.cpp (hunk 1 above)
Comment (MerryMage) [inline Source/Core/AudioCommon/DPL2Decoder.cpp]: Yes, done.

**Comment by MerryMage** [comment]
Other changes made: Edited comment for the hamming window function, <string.h> -> <cstring\>.

**Pull request merged**